src/backend/base/langflow/components/prompts/LangChainHubPrompt.py

```diff
@@ -0,0 +1,107 @@
+    description: str = "Prompt Component that uses LangChain Hub prompts"
+    beta = True
+    icon = "prompts"
+    trace_type = "prompt"
+    name = "LangChain Hub Prompt"
+
+    inputs = [
+        SecretStrInput(
+            name="langchain_api_key",
+            display_name="Your LangChain API Key",
+            info="The LangChain API Key to use.",
+        ),
+        StrInput(
+            name="langchain_hub_prompt",
+            display_name="LangChain Hub Prompt",
+            info="The LangChain Hub prompt to use.",
+            value="efriis/my-first-prompt",
+            refresh_button=True,
+        ),
+    ]
+
+    outputs = [
+        Output(display_name="Build Chat Prompt", name="template", method="build_chat_prompt"),
+    ]
+
+    def update_build_config(self, build_config: dict, field_value: str, field_name: str | None = None):
+        if field_name == "langchain_hub_prompt":
+            template = self._fetch_langchain_hub_template()
+
+            # Extract the messages from the prompt data
+            prompt_template = []
+            for message_data in template.messages:
+                prompt_template.append(message_data.prompt)
+
+            # Regular expression to find all instances of {<string>}
+            pattern = r"\{(.*?)\}"
+
+            # Get all the custom fields
+            custom_fields: List[str] = []
+            full_template = ""
+            for message in prompt_template:
+                # Find all matches
+                matches = re.findall(pattern, message.template)
+                custom_fields = custom_fields + matches
+
+                # Create a string version of the full template
+                full_template = full_template + "\n" + message.template
+
+            # No need to reprocess if we have them already
+            if all(["param_" + custom_field in build_config for custom_field in custom_fields]):
+                return build_config
+
+            # Easter egg: Show template in info popup
+            build_config["langchain_hub_prompt"]["info"] = full_template
+
+            # Remove old parameter inputs if any
+            for key, _ in build_config.copy().items():
+                if key.startswith("param_"):
+                    del build_config[key]
+
+            # Now create inputs for each
+            for custom_field in custom_fields:
+                new_parameter = DefaultPromptField(
+                    name=f"param_{custom_field}",
+                    display_name=custom_field,
+                    info="Fill in the value for {" + custom_field + "}",
+                ).to_dict()
+
+                build_config[f"param_{custom_field}"] = new_parameter
+
+        return build_config
+
+    def build_chat_prompt(
+        self,
+    ) -> ChatPromptTemplate:
+        # Get the parameters that
+        template = self._fetch_langchain_hub_template()  # TODO: doing this twice
+        prompt_value = template.invoke(self._attributes)
```

Comment:
Could you try doing something similar to the Propmpt component at https://github.com/langflow-ai/langflow/blob/2f08abdd6d1b9183fdda5659d5dd5031d13e97c8/src/backend/base/langflow/components/prompts/Prompt.py#L27? 

Basically, we need to pass Message object instead of ChatPromptTemplate. This is causing issues when building the component because self._attributes has keys not required by the prompt and also we'd have compatibility issues with LLM components and others

Reply:
@ogabrielluiz i can do that... but, the templates can contain multiple messages right, from System, User, and others. Are we okay converting that to a single message?

Reply:
The Message is just a vessel. The Prompt object is built inside it.

Reply:
Got it! Just pushed an update @ogabrielluiz